botocore/session.py

```diff
@@ -779,7 +779,9 @@ def create_client(self, service_name, region_name=None, api_version=None,
         event_emitter = self.get_component('event_emitter')
         response_parser_factory = self.get_component(
             'response_parser_factory')
-        if aws_secret_access_key is not None:
+        if config and config.signature_version == 'unsigned':
```

Comment:
Let's use the constant for ``UNSIGNED`` in botocore: https://github.com/boto/botocore/blob/develop/botocore/__init__.py#L54